### ISSUE TITLE
`ramping-arrival-rate` executor shows wrong progressbar 

### ISSUE THREAD

**Issue by na--** [grafana/k6#2951]
### Brief summary

The [`ramping-arrival-rate` scenario executor](https://k6.io/docs/using-k6/scenarios/executors/ramping-arrival-rate/) doesn't handle situations in which the iteration rate doesn't match the duration exactly and iterations finish up very quickly. The progress bar appears unfinished and has a red cross (✗), suggesting that there was a problem, when in fact the executor finished normally.

### k6 version

any since k6 v0.27.0, i.e. this has been a problem since the initial introduction of the `ramping-arrival-rate` executor :sweat_smile:

### OS

Any

### Steps to reproduce the problem

Have a script like this, where the iterations finish quickly and the scenario duration doesn't match the `timeUnit * iteration rate` exactly.
```js
export const options = {
    scenarios: {
        demo: {
            executor: 'ramping-arrival-rate',
            startRate: 1,
            preAllocatedVUs: 5,
            stages: [
                { duration: '2.5s', target: 1 },
            ],
        },
    },
};

export default function () {
    console.log(`iter`);
}

export function handleSummary(data) { }
```

### Expected behaviour

Expect the test run above to finish normally after running exactly 2 iterations.

### Actual behaviour

The test runs exactly 2 iterations, but the progress bar makes it seem like the scenario didn't fully finish or there and was instead prematurely interrupted:
```
...
INFO[0001] iter                                          source=console
INFO[0002] iter                                          source=console

running (02.0s), 0/1 VUs, 2 complete and 0 interrupted iterations
demo ✗ [=============================>--------] 0/1 VUs  2.0s/2.5s  1.00 iters/s
...
```

Basically, it seems like the executor just aborts early, since it has finished all of the 2 iterations it was supposed to execute and it "knows" it's not going to have time to start that third iteration. That is normal and good and desired. However, it doesn't seem to update its progressbar correctly :disappointed: 

One way to "fix" the progress bars is to force the executor it to wait, i.e. slow down the iterations. If I just add `sleep(2)` in the iteration, and then the progress bar is displayed correctly. But so far this seems like just a visualization issue, the execution seems to be working as expected :relieved:
